modules/core/src/main/java/org/apache/ignite/internal/processors/query/stat/task/GatherPartitionStatistics.java

```diff
@@ -318,6 +320,44 @@ private ObjectPartitionStatisticsImpl recollectPartition(
         return res;
     }
 
+    /**
+     * @param cctx Cache contex.
+     * @param desc Row descriptor.
+     * @param row Cache data row
+     * @param coll Column collector.
+     * @return IndexKey containing value extracted from row.
+     */
+    private Object getValue(
+        GridCacheContext<?, ?> cctx,
+        GridQueryRowDescriptor desc,
+        CacheDataRow row,
+        ColumnStatisticsCollector coll
+    ) {
+        if (row.value() == null)
+            return unwrap(cctx, row.key(), desc.type().keyClass());
+
+        if (desc.isKeyColumn(coll.columnId()))
```

Comment:
Perhaps we can use GridQueryProperty instead of additional mapping through columnId, it will simplify the code and avoid duplication of key/val checking.